e2e/authorization_test.go

```diff
@@ -23,135 +23,207 @@ import (
-			Name:      "tls-fail",
-			Namespace: "default",
-		},
-		Spec: monitoringv1.PodMonitoringSpec{
-			Selector: metav1.LabelSelector{
-				MatchLabels: map[string]string{
-					"app": "go-synthetic",
-				},
-			},
-			Endpoints: []monitoringv1.ScrapeEndpoint{
-				{
-					Scheme:   "https",
-					Port:     intstr.FromString("web"),
-					Interval: "5s",
+	t.Run("patch-example-app-args", testPatchExampleAppArgs(ctx, kubeClient, []string{
+		"--basic-auth-username=user",
+	}))
+	authorizationClusterPodMonitoringTest(ctx, t, kubeClient, opClient, "basic-auth-no-password",
+		&monitoringv1.ScrapeEndpoint{
+			Port:     intstr.FromString("web"),
+			Interval: "5s",
+			HTTPClientConfig: monitoringv1.HTTPClientConfig{
+				BasicAuth: &monitoringv1.BasicAuth{
+					Username: "user",
 				},
 			},
+		}, &monitoringv1.ScrapeEndpoint{
+			Port:     intstr.FromString("web"),
+			Interval: "5s",
 		},
-	}
-	errMsg := "x509: certificate signed by unknown authority"
-	t.Run("tls-podmonitoring-failure", testEnsurePodMonitoringFailure(ctx, opClient, pmFail, errMsg))
+		errUnauthorized,
+	)
 }
 
-func TestTLSClusterPodMonitoring(t *testing.T) {
+func TestBasicAuthPodMonitoringNoUsername(t *testing.T) {
 	ctx := context.Background()
 	kubeClient, opClient, err := setupCluster(ctx, t)
 	if err != nil {
 		t.Fatalf("error instantiating clients. err: %s", err)
 	}
 
-	t.Run("collector-deployed", testCollectorDeployed(ctx, kubeClient))
-	t.Run("enable-target-status", testEnableTargetStatus(ctx, opClient))
-	t.Run("patch-example-app-args", testPatchExampleAppArgs(ctx, kubeClient, []string{"--tls-create-self-signed=true"}))
+	const secretName = "basic-auth-no-username"
+	const secretKey = "k1"
+	if err := addSecret(ctx, kubeClient, operator.DefaultOperatorNamespace, metav1.NamespaceDefault, secretName, secretKey, []byte("pass")); err != nil {
+		t.Fatalf("error adding secret: %s", err)
+	}
 
-	cpm := &monitoringv1.ClusterPodMonitoring{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "tls-ready",
-			Namespace: "default",
-		},
-		Spec: monitoringv1.ClusterPodMonitoringSpec{
-			Selector: metav1.LabelSelector{
-				MatchLabels: map[string]string{
-					"app": "go-synthetic",
-				},
-			},
-			Endpoints: []monitoringv1.ScrapeEndpoint{
-				{
-					Scheme:   "https",
-					Port:     intstr.FromString("web"),
-					Interval: "5s",
-					HTTPClientConfig: monitoringv1.HTTPClientConfig{
-						TLS: &monitoringv1.TLS{
-							InsecureSkipVerify: true,
+	t.Run("patch-example-app-args", testPatchExampleAppArgs(ctx, kubeClient, []string{
+		"--basic-auth-password=pass",
+	}))
+	authorizationPodMonitoringTest(ctx, t, kubeClient, opClient, "basic-auth-no-username",
+		&monitoringv1.ScrapeEndpoint{
+			Port:     intstr.FromString("web"),
+			Interval: "5s",
+			HTTPClientConfig: monitoringv1.HTTPClientConfig{
+				BasicAuth: &monitoringv1.BasicAuth{
+					Password: &monitoringv1.SecretSelector{
+						KubernetesSecret: &monitoringv1.KubernetesSecretKeySelector{
```

Comment:
hmmmmm, I thought we renamed to secret, should we rename the type quickly, while we are renaming CR field?